api/src/main/java/io/kafbat/ui/config/auth/BasicAuthSecurityConfig.java

```diff
@@ -42,4 +58,39 @@ public SecurityWebFilterChain configure(ServerHttpSecurity http) {
     return builder.build();
   }
 
+  @Bean
+  public ReactiveUserDetailsService reactiveUserDetailsService(SecurityProperties properties,
+                                                               ObjectProvider<PasswordEncoder> passwordEncoder,
+                                                               AccessControlService accessControlService) {
+    SecurityProperties.User user = properties.getUser();
+
+    UserDetails userDetails = User.withUsername(user.getName())
+        .password(password(user.getPassword(), passwordEncoder.getIfAvailable()))
+        .roles(StringUtils.toStringArray(user.getRoles()))
+        .build();
+
+    if (accessControlService.isRbacEnabled()) {
+      RbacBasicAuthAuthoritiesExtractor extractor = new RbacBasicAuthAuthoritiesExtractor(accessControlService);
+
+      return new RbacUserDetailsService(new RbacBasicAuthUser(userDetails, extractor.groups(user.getName())));
+    } else {
+      return new MapReactiveUserDetailsService(userDetails);
+    }
+  }
+
+  private String password(String password, PasswordEncoder encoder) {
```

Comment:
I don't think we even need this method, we could delegate this to the encoder which should handle this via `encoder.encode(password);`